packages/shelter/src/util.ts

```diff
@@ -86,6 +86,65 @@ export function createSubscription<TState, TStoreData = any>(
   return data;
 }
 
+
+const storeInitPromises = new WeakMap<FluxStore, Promise<void>>();
+
+// awaits until the _isInitialized property of a store is true by overwritting it's setter
+function awaitStoreInit(store: FluxStore): Promise<void> {
+  return storeInitPromises[store] ?? (storeInitPromises[store] = new Promise<void>(resolve => {
+    if (store._isInitialized) resolve();
```

Comment:
```suggestion
    if (store._isInitialized) return resolve();
```

I assume the intention here is to bail out, as we know that this condition is already fulfilled, however the rest of this function continues executing, adding the property etc.

Regarding the suggested change - we don't care about the return value of `resolve()` (its just `undefined`, not that that even matters!), but it explicitly stops any unnecessary side effects now.